### ISSUE TITLE
Install x86_64 JVM on M1 mac

### ISSUE THREAD

**Issue by nedtwigg** [halcyon/asdf-java#156]
I've got an M1 mac. Sometimes I need an aarch64-native JVM, but othertimes I need an x86_64 JVM under rosetta translation. It would be great for asdf-java to support this. As best I can tell, I can't do that right now.

This M1 arm/rosetta divide was discussed a bit at https://github.com/asdf-vm/asdf/issues/834, where it was suggested that one solution is for plugins to encode CPU architecture in the version string.

If I understand correctly, this plugin offers one set of JVMs on an Intel machine, and a different set of JVMs on Apple Silicon (e.g. no AdoptOpenJDK on Apple Silicon). What do you think about making the Intel JVMs available to Apple Silicon machines, but under a `-rosetta` version string (or some other name)?

**Comment by nunezger** [comment]
Yes, they are different sets of JVM depending on the underlying architecture. The plugin stores the set of JVM releases in a cache file which then gets used for all subsequent asdf java plugin commands. I made a change to make the cache file platform specific by adding the OS and architecture to the name of the cache file

**Comment by bayeslearner** [comment]
You can get around this by creating a profile in the terminal app for running bash in rosetta, then using asdf to install the x86 version of the binary in a new window of that profile. 
<img width="449" alt="image" src="https://user-images.githubusercontent.com/68746303/196815525-19e73bbd-fce3-4915-acae-6c4032a4d1da.png">

**Comment by leonard84** [comment]
The readme mentions this workaround:

> To switch your terminal from native ARM to Rosetta use arch -x86_64 /bin/zsh.

Howerver, the [architecture detection](https://github.com/halcyon/asdf-java/blob/8155559d75e0f153e4ec575917cd59bb405e97f3/bin/functions#L96C1-L103) in the script isn't affected by that anymore. When starting a shell this way `arch` reports `i386` but `uname -m` reports `arm64`.

**Comment by RafaelMoreira1180778** [comment]
> The readme mentions this workaround:
> 
> > To switch your terminal from native ARM to Rosetta use arch -x86_64 /bin/zsh.
> 
> Howerver, the [architecture detection](https://github.com/halcyon/asdf-java/blob/8155559d75e0f153e4ec575917cd59bb405e97f3/bin/functions#L96C1-L103) in the script isn't affected by that anymore. When starting a shell this way `arch` reports `i386` but `uname -m` reports `arm64`.

This is problematic because:

```shell
$ env /usr/bin/arch -x86_64 /bin/zsh --login

$ arch
i386

$ uname -m
arm64

$ asdf install java adoptopenjdk-11.0.21+9
############################################################################################################################################################################################## 100.0%
OpenJDK11U-jdk_aarch64_mac_hotspot_11.0.21_9.tar.gz
OpenJDK11U-jdk_aarch64_mac_hotspot_11.0.21_9.tar.gz: OK
```

Notice the architecture that was installed, because of the difference between `uname -m` and `arch`.

I believe this should be changed!

**Comment by joshjprice** [comment]
Although the comment is almost two years old @RafaelMoreira1180778 has a very good point.

I experience the inverse problem where asdf consistently installs x86_64 JDK's on my M3 Macbook.
I am not running the terminal under rosetta. I get 

`
arch
arm64
`
and
`
uname -m
arm64
`

in the terminal, but I get only x86_64 JDKs installed so far.

I picked a case where I know an arm64 jdk is available because I manually downloaded it. I figured that was important to test because asdf might download an x86 jdk if no arm64 jdk is available.

When we combine this with the fact that symlinks to manually installed jdks are no longer working (reported in [another issue](https://github.com/asdf-vm/asdf/issues/1873)), it's near impossible for me to use asdf for java.
